CONFIGURATION.md

```diff
@@ -181,3 +183,38 @@ end
 ```
 
 Note that with Rails >= 7.1 the comment format can be specified via the `config.active_record.query_log_tags_format` option. SolarWinds Observability functionality depends on the default `:sqlcommenter` format, it is not recommended to change this value.
+
+### Background Jobs
+
+#### Resque
+
+[Resque](https://github.com/resque/resque) is a Redis-backed library for creating background jobs, queuing them on multiple queues, and processing them later.
+
+When starting the Resque worker, it is necessary to set two options: `SW_APM_ENABLE_AFTER_FORK=true` and `RUN_AT_EXIT_HOOKS=1`.
+
+For example:
+
+```console
+SW_APM_ENABLE_AFTER_FORK=true RUN_AT_EXIT_HOOKS=1 QUEUE=${QUEUE_NAME} ${EXTRA_OPTIONS} bundle exec rake resque:work
+```
+
+Explanation:
+
+* `SW_APM_ENABLE_AFTER_FORK`: This option enables the `solarwinds_apm` library to defer starting a new reporter for each forked process created by the Resque worker. This is crucial to avoid segmentation faults.
+* `RUN_AT_EXIT_HOOKS`: This option, provided by Resque, ensures that the forked processes shut down gracefully (i.e., no immediate `exit!`).
+
+Additionally, you need to configure the Resque initializer in your Rails application by adding the following code to `config/initializers/resque.rb`:
+
+```ruby
+require 'resque'
+
+Resque.after_fork do
+  while !SolarWindsAPM::API.solarwinds_ready?(1_000)
+    puts "Waiting"
+  end
```

Comment:
Might have misled you with how i stated this, but looking at [underlying liboboe code](https://github.com/solarwinds-cloud/solarwinds-apm-liboboe/blob/defer_init_on_top_of_15.0.2/liboboe/settings.cpp#L1607) it will return as soon as "ready", so the check via Ruby SDK call can simply be something like:
```
unless SolarWindsAPM::API.solarwinds_ready?(8_000)
  puts "SolarWindsAPM is not ready, this job will not be traced."
end
```

cc @jerrytfleung to confirm this is behaviour for current and future liboboe versions.

Reply:
Yes, this makes more sense; updated